pkg/plugins/policies/meshtimeout/plugin/xds/configurer.go

```diff
@@ -218,3 +221,77 @@ func toProtoDurationOrDefault(d *kube_meta.Duration, defaultDuration time.Durati
 	}
 	return util_proto.Duration(d.Duration)
 }
+
+var DefaultTimeoutConf = api.Conf{
+	ConnectionTimeout: &kube_meta.Duration{Duration: policies_defaults.DefaultConnectTimeout},
+	IdleTimeout:       &kube_meta.Duration{Duration: policies_defaults.DefaultIdleTimeout},
+	Http: &api.Http{
+		RequestTimeout:        &kube_meta.Duration{Duration: policies_defaults.DefaultRequestTimeout},
+		StreamIdleTimeout:     &kube_meta.Duration{Duration: policies_defaults.DefaultGatewayStreamIdleTimeout},
+		MaxStreamDuration:     &kube_meta.Duration{Duration: policies_defaults.DefaultMaxStreamDuration},
+		MaxConnectionDuration: &kube_meta.Duration{Duration: policies_defaults.DefaultConnectTimeout},
+		RequestHeadersTimeout: &kube_meta.Duration{Duration: policies_defaults.DefaultRequestHeadersTimeout},
+	},
+}
+
+type ListenerConfigurer struct {
+	Conf     api.Conf
+	Protocol core_mesh.Protocol
+}
+
+func (rc *ListenerConfigurer) ConfigureListener(listener *envoy_listener.Listener) error {
+	if listener == nil {
+		return nil
+	}
+
+	httpTimeouts := func(hcm *envoy_hcm.HttpConnectionManager) error {
+		rc.configureRequestTimeout(hcm.GetRouteConfig())
+		rc.configureRequestHeadersTimeout(hcm)
+		// old Timeout policy configures idleTimeout on listener while MeshTimeout sets this in cluster
+		if hcm.CommonHttpProtocolOptions == nil {
+			hcm.CommonHttpProtocolOptions = &envoy_core.HttpProtocolOptions{}
+		}
+
+		hcm.CommonHttpProtocolOptions.IdleTimeout = util_proto.Duration(0)
+		return nil
+	}
+	tcpTimeouts := func(proxy *envoy_tcp.TcpProxy) error {
```

Comment:
do we need to duplicate all of this?

Reply:
I tried to reuse some of the code, but with MeshService, Mesh*Routes and how we target real resources code will have small differences. But we will remove the old configurer after we migrate to MeshService